libs/checkpoint-duckdb/langgraph/checkpoint/duckdb/base.py

```diff
@@ -0,0 +1,290 @@
+import json
+import random
+from typing import Any, List, Optional, Sequence, Tuple, cast
+
+from langchain_core.runnables import RunnableConfig
+
+from langgraph.checkpoint.base import (
+    WRITES_IDX_MAP,
+    BaseCheckpointSaver,
+    ChannelVersions,
+    Checkpoint,
+    CheckpointMetadata,
+    get_checkpoint_id,
+)
+from langgraph.checkpoint.serde.jsonplus import JsonPlusSerializer
+from langgraph.checkpoint.serde.types import TASKS, ChannelProtocol
+
+MetadataInput = Optional[dict[str, Any]]
+
+"""
+To add a new migration, add a new string to the MIGRATIONS list.
+The position of the migration in the list is the version number.
+"""
+MIGRATIONS = [
+    """CREATE TABLE IF NOT EXISTS checkpoint_migrations (
+    v INTEGER PRIMARY KEY
+);""",
+    """CREATE TABLE IF NOT EXISTS checkpoints (
```

Comment:
Potential issue. The schema for `checkpoints` that I'm creating for the API implementation looks like this:
```sql
CREATE TABLE IF NOT EXISTS checkpoints (
    thread_id UUID NOT NULL,
    checkpoint_id UUID NOT NULL,
    run_id UUID,
    parent_checkpoint_id UUID,
    checkpoint JSON NOT NULL,
    metadata JSON DEFAULT '{}' NOT NULL,
    checkpoint_ns TEXT DEFAULT '' NOT NULL,
    PRIMARY KEY (thread_id, checkpoint_ns, checkpoint_id),
    FOREIGN KEY (run_id) REFERENCES run(run_id)
);
```
This schema doesn't include `run_id` and includes `checkpoint`.

Reply:
Also, I'm using `UUID` instead of `TEXT` for the ID types.

Reply:
discussed offline! will keep the schema as is